### PR TITLE
[Docs] [AutoDiff] Fix incorrectly rendered table and dead link.

### PR DESCRIPTION
**Pull request by rxwei**

### CODE REVIEW

**Review by marcrasi — APPROVED** [review]

**Comment by rxwei** [comment]
@swift-ci please smoke test and merge

File: docs/DifferentiableProgramming.md [hunk 1]
```diff
@@ -2016,10 +2016,11 @@ differentiating either one will provide derivatives with respect to parameters
 `T` and `U`. Here are some examples of first-order function types and their
 corresponding curried function types:
 
-| First-order function type                 | Curried function type                           |
-| @differentiable (T, U) -> V               | @differentiable (T) -> @differentiable (U) -> V |
-| @differentiable (T, @noDerivative U) -> V | @differentiable (T) -> (U) -> V                 |
-| @differentiable (@noDerivative T, U) -> V | (T) -> @differentiable (U) -> V                 |
+| First-order function type                   | Curried function type                             |
+|---------------------------------------------+---------------------------------------------------|
```
Comment (dan-zheng) [inline docs/DifferentiableProgramming.md]: The table still doesn't appear to be rendered correctly. I think this should fix it:
```suggestion
|---------------------------------------------|---------------------------------------------------|
```

Here's what it looks like:

| First-order function type                   | Curried function type                             |
|---------------------------------------------|---------------------------------------------------|
| `@differentiable (T, U) -> V`               | `@differentiable (T) -> @differentiable (U) -> V` |
| `@differentiable (T, @noDerivative U) -> V` | `@differentiable (T) -> (U) -> V`                 |
| `@differentiable (@noDerivative T, U) -> V` | `(T) -> @differentiable (U) -> V`                 |

Comment (rxwei) [reply docs/DifferentiableProgramming.md]: Ugh, emacs is doing weird stuff here. Thanks for pointing this out.

**Comment by rxwei** [comment]
@swift-ci please smoke test and merge

**Comment by rxwei** [comment]
@swift-ci please smoke test and merge

**Pull request merged**